impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js

```diff
@@ -635,29 +643,34 @@ define(["dojo/_base/declare", "dijit/_WidgetBase", "dijit/_TemplatedMixin", "dij
   buildFilterText: function(filter, prompt) {
     var column = this.datasource.getColumnById(filter.column);
     var friendlyOperator = filter.operator;
-    if (filter.combinationType != pentaho.pda.Column.OPERATOR_TYPES.AND || (filter.operator == pentaho.pda.Column.CONDITION_TYPES.EQUAL && filter.value instanceof Array && filter.value.length > 1)) {
-      switch (filter.combinationType) {
-        case pentaho.pda.Column.OPERATOR_TYPES.AND:
-          friendlyOperator = this.getLocaleString("FilterCombinationTypeIn");
-          break;
-        case pentaho.pda.Column.OPERATOR_TYPES.AND_NOT:
-          friendlyOperator = this.getLocaleString("FilterCombinationTypeNotIn");
-          break;
-        default:
-          console.log("Unknown filter combination type for IN condition type: " + filter.combinationType);
-      }
-    } else {
-      // Treat pentaho.pda.Column.DATA_TYPES.UNKNOWN as pentaho.pda.Column.DATA_TYPES.STRING
-      var dataType = column.dataType == pentaho.pda.Column.DATA_TYPES.UNKNOWN ? pentaho.pda.Column.DATA_TYPES.STRING : column.dataType;
-      var comparatorMapping = pentaho.pda.Column.COMPARATOR[dataType];
-      if (comparatorMapping) {
-        array.some(comparatorMapping, function(cArray) {
-          if (cArray[1] === filter.operator) {
-            friendlyOperator = cArray[0];
-            return true;
-          }
-        });
-      }
+    switch (filter.operator) {
+      case pentaho.pda.Column.CONDITION_TYPES.IN:
+        switch (filter.combinationType) {
+          case pentaho.pda.Column.OPERATOR_TYPES.AND:
+            friendlyOperator = this.getLocaleString("FilterCombinationTypeIn");
+            break;
+          case pentaho.pda.Column.OPERATOR_TYPES.AND_NOT:
+            friendlyOperator = this.getLocaleString("FilterCombinationTypeNotIn");
+            break;
+          default:
+            console.log("Unknown filter combination type for IN condition type: " + filter.combinationType);
```

Comment:
careful, in some environments `console` might be `undefined`

Reply:
I used what we already use:
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L283
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L320
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L337
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L357
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L627
https://github.com/pentaho/pentaho-platform-plugin-common-ui/blob/master/impl/client/src/main/javascript/web/dojo/pentaho/common/FilterDialog.js#L647

@afrjorge should I remove all of these?

Reply:
@pentaho/millenniumfalcon ^

Reply:
Those references are in default branches, that apparently should never execute. Maybe that' why they didn't raise any problems.

Wrapping in `typeof` test would solve the problem:
```js
if(typeof console !== "undefined") {
  console.log("message");
}
```